### PR TITLE
View question change

### PR DESCRIPTION
**Pull request by Alleng909**
Work in progress. Removed most routing though random components like expert and highlighting still route using old concept.

### CODE REVIEW

File: src/app/experts/enter-response/enter-response.component.ts
Comment (AdamDKing) [inline src/app/experts/enter-response/enter-response.component.ts]: Let's make the existing newResponse.emit(response) functionality work by asking the backend for an updated list of responses, instead of refreshing via navigation.  It looks like the event is being emitted here but nothing is done with it.

File: src/app/static/filtered-question-list/filtered-question-list.component.ts
Comment (AdamDKing) [inline src/app/static/filtered-question-list/filtered-question-list.component.ts]: Instead of manually calling the lifecycle hook, split the functionality you want out into another method and call that both here and in the hook

File: src/app/static/filtered-question-list/filtered-question-list.component.ts
Comment (AdamDKing) [inline src/app/static/filtered-question-list/filtered-question-list.component.ts]: As discussed, I think let's rework this setup to let preview-questions uncollapse themselves into full view-question components.  It seems a little easier and cleaner than passing information about what questions should be displayed/removed from various components on this page.  Feel free to bring up concerns here, ofc

File: src/app/static/highlighted-response/highlighted-response.component.html
Comment (AdamDKing) [inline src/app/static/highlighted-response/highlighted-response.component.html]: Why change this to id? It looks like you could grab the name the same way as the prior version, and the name is better for users

File: src/app/static/preview-question/preview-question.component.ts
Comment (AdamDKing) [inline src/app/static/preview-question/preview-question.component.ts:27]: ~Please name this something appropriate to what it does.  What does the number it emits represent?~
As mentioned prior, this functionality will probably be reworked, so nevermind!

File: src/app/static/preview-question/preview-question.component.ts
Comment (AdamDKing) [inline src/app/static/preview-question/preview-question.component.ts]: It's possible this is the source of all those 400s from the backend.  Why set questionId in the service to undefined with every preview-question component init?

File: src/app/static/view-question/view-question.component.html
Comment (AdamDKing) [inline src/app/static/view-question/view-question.component.html]: This also applies elsewhere: I'd like a more descriptive name than 'highnum'.  I understand it as the id of the highlighted response, but only because I've been reading this code for awhile.

File: src/app/static/view-question/view-question.component.html
Comment (AdamDKing) [inline src/app/static/view-question/view-question.component.html]: I'm not sure this ngOnChanges(null) does anything?  ngOnChanges is the hook called when changes happen, not how we make the change detector run

File: src/app/static/view-question/view-question.component.ts
Comment (AdamDKing) [inline src/app/static/view-question/view-question.component.ts]: As before, I'm not sure why this ngOnChanges is here or what it's doing in this case

File: src/app/static/view-question/view-question.component.ts
Comment (AdamDKing) [inline src/app/static/view-question/view-question.component.ts:69]: While this logic could use some work, this component might be the right place for actually getting the highlighted response.  Let's put a pin in that functionality and come back to it after the larger rework is done.

**Review by AdamDKing — CHANGES_REQUESTED** [review]
Some changes requested, check out the comments.  I think things will become clearer as we refactor.  The assorted cleanups along the way are nice!

File: src/app/static/highlighted-response/highlighted-response.component.html
Comment (Alleng909) [reply src/app/static/highlighted-response/highlighted-response.component.html]: This was a mistake, I just wanted to remove extra variables.

File: src/app/static/preview-question/preview-question.component.ts
Comment (Alleng909) [reply src/app/static/preview-question/preview-question.component.ts]: I don't even know what questionIdService is used for, so I just removed it completely from this component.

File: src/app/experts/enter-response/enter-response.component.ts
Comment (AdamDKing) [inline src/app/experts/enter-response/enter-response.component.ts]: please remove comment that no longer describes functionality

**Review by AdamDKing — DISMISSED** [review]
The only review comment is asking to remove some outdated comments, so this is approved, nice!

**Review by AdamDKing — DISMISSED** [review]
These changes are functional but not all I'd like to have done -- for another sprint!

**Review by AdamDKing — APPROVED** [review]
Approving after merge

**Pull request merged**
